src/posthog-core.ts

```diff
@@ -1112,13 +1130,15 @@ export class PostHog {
             // let the reload feature flag request know to send this previous distinct id
             // for flag consistency
             this.featureFlags.setAnonymousDistinctId(previous_distinct_id)
-        } else {
-            if (userPropertiesToSet) {
-                this.people.set(userPropertiesToSet)
-            }
-            if (userPropertiesToSetOnce) {
-                this.people.set_once(userPropertiesToSetOnce)
-            }
+        } else if (userPropertiesToSet || userPropertiesToSetOnce) {
+            // If the distinct_id is not changing, but we have user properties to set, send an $identify event
+            this.capture(
+                '$identify',
```

Comment:
This is good, we should unify these into a single call, but we can just call this event `$set` and that should work well without causing any `$identify` side-effects now or in the future

Reply:
For example define a function (feel free to ignore unset part):
```
_set_properties(userPropertiesToSet, userPropertiesToSetOnce, userPropertiesToUnset): void {
  this.capture('$set', {'$set': userPropertiesToSet, '$set_once': userPropertiesToSetOnce, '$unset': userPropertiesToUnset})
}
```
& then from here call it as `this._set_properties(userPropertiesToSet, userPropertiesToSetOnce)`